src/Log/SafeSemanticLogger.php

```
@@ -0,0 +1,139 @@
+    /**
+     * Serialize without session state (no live log carried across serialization)
+     *
+     * @return array<string, mixed>
+     */
+    public function __serialize(): array
+    {
+        return [];
+    }
+
+    /** @param array<string, mixed> $data */
+    public function __unserialize(array $data): void
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$data'. (undefined)

(UnusedFormalParameter)